DirectX11/HackerDXGI.cpp

```diff
@@ -1891,6 +2061,38 @@ STDMETHODIMP HackerDXGISwapChain::SetFullscreenState(THIS_
 	LogInfo("  Fullscreen = %d\n", Fullscreen);
 	LogInfo("  Target = %p\n", pTarget);
 
+	HRESULT hr;
+
+	if (mFakeBackBuffer || mFakeSwapChain) // we need fullscreen anyway
+	{
+		BOOL fst = FALSE;
+		IDXGIOutput *target = nullptr;
+		mOrigSwapChain->GetFullscreenState(&fst, &target);
+
+		if (target)
+			target->Release();
+
+		// dont call setfullscreenstate again to avoid starting mode switching and flooding winproc with unnecessary messages
+		// can disable fullscreen mode somehow
+		if (fst && Fullscreen)
```

Comment:
The full_screen flag is ignored as soon as upscaling is on. The fullscreen check is necessary to avoid state changes in direct x. Somehow if fullscreen is on and you call the game calls the function again the screen goes to windowed mode on some games(witcher). I could change it the way full_screen flag is used, but it would do the whole configuration more complicated. Moreover i have designed the upscaling to be always used with the fullscreen. Thats why i force migoto to ignore old resizing functionality if upscaling is on. This way the user desides whether to use "old" resizing or upscaling but in fullscreen.